### PR TITLE
feat: replace info block directive with mdx component

### PR DESCRIPTION
**Pull request by iamnamananand996**
## Because
Currently, we are using `info-block directive` in markdown to implement components like `info, tip, danger, warning` and need to be replaced with mdx component, will close this issue [https://github.com/instill-ai/instill.tech/issues/496](https://github.com/instill-ai/instill.tech/issues/496)
## This commit
have changes with example added here.

current info-block implementation is replaced now with
```md
:::info{type=info}
This is info block with info type
:::
```
bellow example
- example: https://github.com/instill-ai/instill.tech/issues/496#issuecomment-1455006008
- library to parse .mdx: https://github.com/instill-ai/instill.tech/issues/496#issuecomment-1455042602
- and update documentation also: https://github.com/instill-ai/instill.tech/blob/39211ce41e64b62f629faf503619df7dc37f4719/CONTRIBUTING.md?plain=1#L61

### CODE REVIEW

**Comment by iamnamananand996** [comment]
Hi @EiffelFly,

I have create this pull request, don't know why the checks are failing.
any suggestion you can add will be very helpful.

**Comment by EiffelFly** [comment]
@iamnamananand996 Currently, we won't run the action in the contributor's PR. The reason is this may expose our company-wise secret. We are still investigating a solution around this issue, stay tune. We will come up with a better way to deal with it.

ref: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

**Comment by iamnamananand996** [comment]
okay sure, thank you @EiffelFly

**Comment by EiffelFly** [comment]
@iamnamananand996 Could you help me one more thing first? We want to have the same structure of the PR description

## Because

<describe the reason, you could link the issue here>

## This commit

<describe what you have done under the hood>

**Comment by iamnamananand996** [comment]
updated the PR description now.
cc - @EiffelFly

File: src/components/ui/MdxInfoBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { ReactElement } from "react-markdown/lib/react-markdown";
+
+export type MdxInfoBlockProps = {
+  type?: string;
```
Comment (EiffelFly) [inline src/components/ui/MdxInfoBlock.tsx:4]: @iamnamananand996 type should be required 

```suggestion
type: string
```

File: src/components/ui/MdxInfoBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { ReactElement } from "react-markdown/lib/react-markdown";
```
Comment (EiffelFly) [inline src/components/ui/MdxInfoBlock.tsx:1]: We can directly use ReactElement from react here

```suggestion
import { ReactElement } from "react";
```

**Review by EiffelFly — COMMENTED** [review]
@iamnamananand996 After some investigation, the mdx already treat the markdown in the jsx as first class citizen so it will parse them to markdown ([ref](https://mdxjs.com/docs/what-is-mdx/#markdown-for-the-component-era)). We don't need to leverage react-markdown here.

We can directly use the ReactElement type provided by React.

I had added some review! Please help us correct them, Thanks!

**Comment by EiffelFly** [comment]
@xiaofei-du Because this PR has some side-effect to the docs, please help us review the changes.

**Comment by iamnamananand996** [comment]
updated the request changes.
cc - @EiffelFly

**Review by xiaofei-du — APPROVED** [review]
Thanks @iamnamananand996 for the contribution! Looks good!

**Pull request merged**